systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java

```diff
@@ -0,0 +1,133 @@
+    public void testCertificates() {
+        LOGGER.info("Running testCertificates {}", CLUSTER_NAME);
+
+        String commandForKafkaBootstrap = "openssl s_client -connect my-cluster-kafka-bootstrap:9093 -showcerts " +
+                        "-CAfile /opt/kafka/broker-certs/cluster-ca.crt " +
+                        "-verify_hostname my-cluster-kafka-bootstrap";
+        String outputForKafkaBootstrap =
+                kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, 0), "kafka", "/bin/bash", "-c", commandForKafkaBootstrap).out();
+        checkKafkaCertificates(outputForKafkaBootstrap);
+
+
+        String commandForZookeeperClient = "openssl s_client -connect my-cluster-zookeeper-client:2181 -showcerts " +
+                "-CAfile /opt/kafka/broker-certs/cluster-ca.crt " +
+                "-verify_hostname my-cluster-zookeeper-client";
+        String outputForZookeeperClient =
+                kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, 0), "kafka", "/bin/bash", "-c", commandForZookeeperClient).out();
+        checkZookeeperCertificates(outputForZookeeperClient);
+
+        IntStream.rangeClosed(0, 1).forEach(podId -> {
+            String commandForKafkaPort9091 = generateOpenSSLCommandWithCerts(kafkaPodName(CLUSTER_NAME, podId), "my-cluster-kafka-brokers", "9091");
+            String commandForKafkaPort9093 = generateOpenSSLCommandWithCAfile(kafkaPodName(CLUSTER_NAME, podId), "my-cluster-kafka-brokers", "9093");
+
+            String outputForKafkaPort9091 =
+                    kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, podId), "kafka", "/bin/bash", "-c", commandForKafkaPort9091).out();
+            String outputForKafkaPort9093 =
+                    kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, podId), "kafka", "/bin/bash", "-c", commandForKafkaPort9093).out();
+
+            checkKafkaCertificates(outputForKafkaPort9091, outputForKafkaPort9093);
+
+            String commandForZookeeperPort2181 = generateOpenSSLCommandWithCAfile(zookeeperPodName(CLUSTER_NAME, podId), "my-cluster-zookeeper-nodes", "2181");
+            String commandForZookeeperPort2888 = generateOpenSSLCommandWithCerts(zookeeperPodName(CLUSTER_NAME, podId), "my-cluster-zookeeper-nodes", "2888");
+            String commandForZookeeperPort3888 = generateOpenSSLCommandWithCerts(zookeeperPodName(CLUSTER_NAME, podId), "my-cluster-zookeeper-nodes", "3888");
+
+            String outputForZookeeperPort2181 =
+                    kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, podId), "kafka", "/bin/bash", "-c", commandForZookeeperPort2181).out();
+            checkZookeeperCertificates(outputForZookeeperPort2181);
+
+            try {
+                String outputForZookeeperPort2888 =
+                        kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, podId), "kafka", "/bin/bash", "-c", commandForZookeeperPort2888).out();
+                String outputForZookeeperPort3888 =
```

Comment:
maybe the test against 3888 can be outside of the try/catch, we faced the problem on 2888 only right?

Reply:
Oh, yes!